streamer/src/nonblocking/quic.rs

```diff
@@ -1184,22 +1172,6 @@ async fn handle_chunk(
 ) -> bool {
     if let Some(chunk) = maybe_chunk {
         trace!("got chunk: {:?}", chunk);
-        let chunk_len = chunk.bytes.len() as u64;
```

Comment:
It's safe to remove these checks because:

- we set the stream window size to PACKET_DATA_SIZE, so if a peer sends more, they get automatically disconnected and the stream is reset
- every time we read a chunk, we check that accum.meta.size <= PACKET_DATA_SIZE, so even if we introduce a bug where we set the window larger (which will never happen, but just in case)
- the overflow checks aren't needed since PACKET_DATA_SIZE is a small number and we always check we're below that

Reply:
<img width="1612" alt="Screenshot 2024-10-23 at 3 04 20 PM" src="https://github.com/user-attachments/assets/86ad91bc-5a29-4d5c-976b-64ea4da2c4a4">
Stats on these

Reply:
> I think we should retain this safe checks I do not see huge perf impact.

this is not about perf, it's about making things that make sense.

we call 

```
let chunk = match tokio::select! {
                    chunk = tokio::time::timeout(
                        wait_for_chunk_timeout,
                        stream.read_chunk(PACKET_DATA_SIZE, true)) => chunk,

                    // If the peer gets disconnected stop the task right away.
                    _ = cancel.cancelled() => break,
                }
```

So we say we're willing to read max PACKET_DATA_SIZE. 

Here's the implementation of the future returned by quinn https://github.com/quinn-rs/quinn/blob/3f2e1aaca6a80ad96e26e05c996b8e07db25257a/quinn/src/recv_stream.rs#L670

max_length is passed to chunks.next https://github.com/quinn-rs/quinn/blob/3f2e1aaca6a80ad96e26e05c996b8e07db25257a/quinn/src/recv_stream.rs#L184

max_length is passed to rs.assembler.read https://github.com/quinn-rs/quinn/blob/3f2e1aaca6a80ad96e26e05c996b8e07db25257a/quinn-proto/src/connection/streams/recv.rs#L298

read never returns more than max_length https://github.com/quinn-rs/quinn/blob/3f2e1aaca6a80ad96e26e05c996b8e07db25257a/quinn-proto/src/connection/assembler.rs#L90

So, what's your explanation for the metric being != 0? Mine is that there's some broken validator in the wild where we were doing the wrong thing that is logging the metric. But now we're doing the right thing so we don't need to log this metric. In the PR there's a debug_assert and an error!() that gets logged if the size is wrong, surely that's enough?

Reply:
If you filter on 3gGnnwaCHuGaBzhAa8ujFR1vf6inHYPxLBh8tdXHTxgZ or la1ne or any of the good validators, the metric is 0, corroborating that this must be some old broken validator version.

Reply:
I see that if a peer sends up invalid streams > PACKET_DATA_SIZE, we keep reading streams from that connection. I think we should terminate the connection instead, I'll send that in a separate PR.

Reply:
Ok what I think is happening is this

```
if chunk.offset > PACKET_DATA_SIZE as u64 || chunk_len > PACKET_DATA_SIZE as u64 {
                    stats.total_invalid_chunks.fetch_add(1, Ordering::Relaxed);
                    return true;
                }
```

This will increase total_invalid_chunks if chunk.offset > PACKET_DATA_SIZE, that is if someone sends a transaction > PACKET_DATA_SIZE. 

As per comment above, I think we should disconnect when that happens. I'll put back a metric called "invalid_stream_size" or something